PythonAPI/carla/agents/navigation/local_planner.py

```diff
@@ -199,7 +199,7 @@ def run_step(self, debug=True):
         if not self._global_plan and len(self._waypoints_queue) < int(self._waypoints_queue.maxlen * 0.5):
             self._compute_next_waypoints(k=100)
 
-        if len(self._waypoints_queue) == 0:
+        if len(self._waypoints_queue) == 0 and len(self._waypoint_buffer) == 0:
```

Comment:
Agreed :) This is definitely the right way to go. Thanks for pointing this out, I had missed it.